cdap-examples/DataCleansing/src/main/java/co/cask/cdap/examples/datacleansing/DataCleansingMapReduce.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.examples.datacleansing;
+
+import co.cask.cdap.api.Resources;
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.BatchPartitionConsumer;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.api.dataset.lib.PartitionKey;
+import co.cask.cdap.api.dataset.lib.PartitionedFileSet;
+import co.cask.cdap.api.dataset.lib.PartitionedFileSetArguments;
+import co.cask.cdap.api.mapreduce.AbstractMapReduce;
+import co.cask.cdap.api.mapreduce.MapReduceContext;
+import co.cask.cdap.api.metrics.Metrics;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.Reducer;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * A simple MapReduce that reads records from the inputRecords PartitionedFileSet and writes all records
+ * that match a particular {@link Schema} to the outputRecords PartitionedFileSet. It also keeps track of its state of
+ * which partitions it has processed, so that it only processes new partitions of data each time it runs.
+ */
+public class DataCleansingMapReduce extends AbstractMapReduce {
+  protected static final String NAME = "DataCleansingMapReduce";
+  protected static final String OUTPUT_PARTITION_TIME = "output.partition.time";
+
+  private final BatchPartitionConsumer partitionConsumer = new BatchPartitionConsumer() {
+    private static final String STATE_KEY = "state.key";
+
+    @Nullable
+    @Override
+    protected byte[] readBytes(DatasetContext datasetContext) {
+      return ((KeyValueTable) datasetContext.getDataset("consumingState")).read(STATE_KEY);
+    }
+
+    @Override
+    protected void writeBytes(DatasetContext datasetContext, byte[] stateBytes) {
+      ((KeyValueTable) datasetContext.getDataset("consumingState")).write(STATE_KEY, stateBytes);
+    }
+  };
+
+  @Override
+  public void configure() {
+    setName(NAME);
```

Comment:
I think its ok to stick with using constant, since it seems cleaner than getting the class's simplename. Also, many other example apps do it with that style. Either way, it's probably not a big deal.